src/main/java/io/strimzi/kafka/bridge/SourceBridgeEndpoint.java

```diff
@@ -80,11 +82,11 @@ protected void handleClose() {
      */
     protected void send(KafkaProducerRecord<String, byte[]> krecord, Handler<AsyncResult<RecordMetadata>> handler) {
 
-        log.debug("Sending to Kafka on topic {} at partition {} and key {}", krecord.topic(), krecord.partition(), krecord.key());
         if (handler == null) {
             this.producerSettledMode.write(krecord);
         } else {
-            this.producerUnsettledMode.write(krecord, handler);
+            log.debug("Sending to topic " + krecord.topic() + " at partition {}", krecord.partition());
+            this.producerUnsettledMode.exceptionHandler(e -> handler.handle(Future.failedFuture(e))).write(krecord, handler);
```

Comment:
I have just a doubt but you could check easily. Are we sure that the `handler` isn't called twice?
So one time by the exceptionHandler and the other time by the write but with the async result as failed (which is something we check in the handler).